whois-query/src/main/java/net/ripe/db/whois/query/planner/AbuseCFinder.java

```diff
@@ -46,6 +54,26 @@ public AbuseCFinder(@Qualifier("jdbcRpslObjectSlaveDao") final RpslObjectDao obj
         this.ipv6Tree = ipv6Tree;
         this.maintainers = maintainers;
         this.abuseValidationStatusDao = abuseValidationStatusDao;
+        this.ripeSources = getRipeSources(mainSource, nonAuthSource, grsSource);
+    }
+
+    private ImmutableSet<CIString> getRipeSources(final String mainSource, final String nonAuthSource, final String grsSource) {
+        ImmutableSet.Builder<CIString> sourceBuilder  = new ImmutableSet.Builder<>();
+
+        if(StringUtils.isNotEmpty(mainSource)) {
+            sourceBuilder.add(CIString.ciString(mainSource));
+        }
+
+        if(StringUtils.isNotEmpty(nonAuthSource)) {
+            sourceBuilder.add(CIString.ciString(nonAuthSource));
+        }
+
+        final String matchingGrs = mainSource+ "-GRS";
```

Comment:
I'd just automatically add a GRS source as mainsource + "GRS", no need to check if one is configured